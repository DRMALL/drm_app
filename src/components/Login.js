import React, { Component } from 'react'
import { AsyncStorage, View, Text, TextInput, Image, TouchableOpacity, KeyboardAvoidingView, Alert } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import { drmOne, drmTwo, drmThree, loginInputEmail, loginInputWord, loginText, loginForgetWord } from '../common/strings'
import { login } from '../styles'
import Button from './units/Button'
import TextInputImg from './units/TextInputImg'
import { checkToken, depositToken, clearToken } from '../utils/handleToken'
import { postPort } from '../utils/fetchMethod'
import { signIn } from '../apis'


const loginScreenLogo = require('../images/login_screen_logo.png')
const loginPasswordShow = require('../images/login_password_show.png')
const loginPasswordHide = require('../images/login_password_hide.png')

export default class Login extends Component {
  static navigationOptions = {
    headerStyle: {
      height: 0,
      top: 50,
    }
  };

  constructor(props) {
    super(props)
    this.state = { 
      textEmail: '', 
      textWord: '',
    }
  }

  componentDidMount() {
    checkToken('drmAppToken')
    .then(token => {
      if(token) this.props.navigation.navigate('main')
    })
    SplashScreen.hide()
    // clearToken()
  }

  async loginPressButton() {
    let bodyData = {
      email: this.state.textEmail,
      password: this.state.textWord,
    }
    let res = await postPort(signIn, bodyData)
    if(res.code == 201) {
      depositToken('drmAppToken', res.data)
      this.props.navigation.navigate('main')
    } else {
      Alert.alert('❌错误', '邮箱或密码输入有误',
        [
          {text: 'OK', onPress: () => 'OK'},
        ],
        { cancelable: false }
      )
    }
  }

  forgetPressButton() {
    this.props.navigation.navigate('emailVerify')
  }

  render() {
    return (
      <View style={login.wrap}>
        <Image style={login.image} source={loginScreenLogo}/>
        <Text style={login.text}>{drmOne}</Text>
        <Text style={login.text}>{drmTwo}</Text>
        <Text style={[login.text, {lineHeight: 48, marginBottom: 40}]}>{drmThree}</Text>
        <KeyboardAvoidingView behavior={'padding'} >
          <TextInput 
            style={login.textInput}
            placeholder={loginInputEmail}
            onChangeText={(textEmail) => this.setState({textEmail})}
            value={this.state.textEmail}
            // keyboardType='numeric'
            selectTextOnFocus={true}
            blurOnSubmit={true}
            underlineColorAndroid='transparent'
            autoCapitalize='none'
          />
          <TextInputImg 
            viewStyle={{position: 'relative'}}
            inputStyle={login.textInput}
            placeholder={loginInputWord}
            onChangeText={(textWord) => this.setState({textWord})}
            value={this.state.textWord}
            secureTextEntry={true}
            onFocus={this.onFocusInput}
            imgStyle={{width: 25, resizeMode: 'contain', marginBottom: 10 }}
            imgSourceT={loginPasswordShow}
            imgSourceF={loginPasswordHide}
          />
        </KeyboardAvoidingView>
        <Button 
          title={loginText}
          titleStyle={login.touchLoginText}
          style={login.touchOpacity} 
          onPress={this.loginPressButton.bind(this)}
        />
        <Button 
          title={loginForgetWord}
          titleStyle={login.touchForgetText}
          style={{height: 40, margin: 20, justifyContent: "center" }} 
          onPress={this.forgetPressButton.bind(this)}
        />
      </View>
    )
  }
}