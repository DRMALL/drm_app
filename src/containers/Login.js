import React, { Component } from 'react'
import { AsyncStorage, View, Text, TextInput, Image, TouchableOpacity, KeyboardAvoidingView, Alert } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import { NavigationActions } from 'react-navigation'
import { drmOne, drmTwo, drmThree, loginInputEmail, loginInputWord, loginText, loginForgetWord, tokenKey } from '../common/strings'
import { login } from '../styles'
import Button from '../components/units/Button'
import TextInputImg from '../components/units/TextInputImg'
import { checkToken, depositToken, clearToken } from '../utils/handleToken'
import { postPort } from '../utils/fetchMethod'
import { signIn } from '../apis'

import store from '../utils/store'
import changeLoginEmail from '../actions/changeLoginEmail'
import changeLoginWord from '../actions/changeLoginWord'

const loginScreenLogo = require('../images/login_screen_logo.png')
const loginPasswordShow = require('../images/login_password_show.png')
const loginPasswordHide = require('../images/login_password_hide.png')

const resetAction = NavigationActions.reset({
  index: 0,
  actions: [
    NavigationActions.navigate({ routeName: 'main'}),
  ]
})

export default class Login extends Component {
  static navigationOptions = {
    headerStyle: {
      height: 0,
      top: 50,
    }
  };

  constructor(props) {
    super(props)
    this.state = store.getState().login
  }

  componentDidMount() {
    checkToken(tokenKey)
    .then(token => {
      if(token) this.props.navigation.dispatch(resetAction)
    })
    setTimeout(() => {
      SplashScreen.hide()
    }, 2000)
    // clearToken()
  }

  componentWillMount() {
    this.unsubscribe = store.subscribe( ()=> this.setState(store.getState().login) )
  }

  componentWillUnmount(){
    this.unsubscribe()
  }

  async loginPressButton() {
    let bodyData = {
      email: this.state.textEmail,
      password: this.state.textWord,
    }
    let res = await postPort(signIn, bodyData)
    if(res.code == 201) {
      depositToken(tokenKey, res.data)
      this.props.navigation.dispatch(resetAction)
    } else {
      Alert.alert('❌错误', '邮箱或密码输入有误',
        [
          {text: 'OK', onPress: () => 'OK'},
        ],
        { cancelable: false }
      )
    }
  }

  forgetPressButton() {
    this.props.navigation.navigate('emailVerify')
  }

  render() {
    let { textEmail, textWord } = this.state
    return (
      <View style={login.wrap}>
        <Image style={login.image} source={loginScreenLogo}/>
        <Text style={login.text}>{drmOne}</Text>
        <Text style={login.text}>{drmTwo}</Text>
        <Text style={[login.text, {lineHeight: 48, marginBottom: 40}]}>{drmThree}</Text>
        <KeyboardAvoidingView behavior={'padding'} >
          <TextInput 
            style={login.textInput}
            placeholder={loginInputEmail}
            onChangeText={changeLoginEmail}
            value={textEmail}
            // keyboardType='numeric'
            selectTextOnFocus={true}
            blurOnSubmit={true}
            underlineColorAndroid='transparent'
            autoCapitalize='none'
          />
          <TextInputImg 
            viewStyle={{position: 'relative'}}
            inputStyle={login.textInput}
            placeholder={loginInputWord}
            onChangeText={changeLoginWord}
            value={textWord}
            secureTextEntry={true}
            // onFocus={this.onFocusInput}
            imgStyle={{width: 25, resizeMode: 'contain', marginBottom: 10 }}
            imgSourceT={loginPasswordShow}
            imgSourceF={loginPasswordHide}
          />
        </KeyboardAvoidingView>
        <Button 
          title={loginText}
          titleStyle={login.touchLoginText}
          style={login.touchOpacity} 
          onPress={this.loginPressButton.bind(this)}
        />
        <Button 
          title={loginForgetWord}
          titleStyle={login.touchForgetText}
          style={{height: 40, margin: 20, justifyContent: "center" }} 
          onPress={this.forgetPressButton.bind(this)}
        />
      </View>
    )
  }
}