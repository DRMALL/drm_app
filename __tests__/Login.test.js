import 'react-native'
import React from 'react'
import Login from '../src/containers/Login'
import renderer from 'react-test-renderer';

test('Login test', () => {
  const tree = renderer.create(
    <Login />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})