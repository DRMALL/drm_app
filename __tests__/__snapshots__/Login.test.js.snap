// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F7F7F7",
        "height": "110%",
        "justifyContent": "center",
      },
      Object {
        "height": "103.5%",
        "top": -20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 60,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "marginBottom": 20,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "lineHeight": 35,
      }
    }
  >
    多元热流体设备
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "lineHeight": 35,
      }
    }
  >
    信息及用户管理系统
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 20,
          "lineHeight": 35,
        },
        Object {
          "lineHeight": 48,
          "marginBottom": 40,
        },
      ]
    }
  >
    MMHFD
  </Text>
  <ActivityIndicator
    animating={false}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <View
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      blurOnSubmit={true}
      onChangeText={[Function]}
      placeholder="请输入邮箱"
      selectTextOnFocus={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderColor": "#7B7B7B",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 10,
          "opacity": 0.5,
          "paddingLeft": 15,
          "width": 280,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        blurOnSubmit={true}
        onBlur={undefined}
        onChangeText={[Function]}
        onFocus={undefined}
        onSubmitEditing={undefined}
        placeholder="请输入密码"
        secureTextEntry={true}
        selectTextOnFocus={true}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderColor": "#7B7B7B",
            "borderRadius": 6,
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 10,
            "opacity": 0.5,
            "paddingLeft": 15,
            "width": 280,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 7,
            "opacity": 1,
            "position": "absolute",
            "right": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 10,
              "resizeMode": "contain",
              "width": 25,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#579df2",
          "borderRadius": 6,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 280,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 15,
            "overflow": "hidden",
          }
        }
      >
        登录
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 40,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#7B7B7B",
          "fontSize": 15,
          "opacity": 0.5,
        }
      }
    >
      忘记密码?
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "height": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
